client/src/worker_registry/mod.rs

```diff
@@ -157,6 +155,7 @@ impl SlotManager {
 
     #[cfg(test)]
     /// Returns (num_providers, num_buckets), where a bucket key is namespace+task_queue.
+    /// There is only one provider per bucket so `num_providers` should be equal to `num_buckets`.
     pub fn num_providers(&self) -> (usize, usize) {
         self.manager.read().num_providers()
     }
```

Comment:
I think you can just get rid of this entirely now

Reply:
I was checking the invariant that both maps have the same size in the tests, but yes, not much value...

Reply:
Sorry, or you meant just get rid of the comment?

Reply:
I meant the test generally

Reply:
I use it in all the unit tests in that file, I need to check that things are deallocated/allocated properly, and with the wrapping  I can't get to the maps directly...